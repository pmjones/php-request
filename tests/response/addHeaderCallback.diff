001+ Fatal error: Uncaught Error: Call to undefined method SapiResponse::getHeaderCallbacks() in /Users/pmjones/Code/pmjones/php-request/tests/response/addHeaderCallback.php:4
002+ Stack trace:
003+ #0 {main}
004+   thrown in /Users/pmjones/Code/pmjones/php-request/tests/response/addHeaderCallback.php on line 4
001- bool(true)
002- bool(false)
003- int(1)
004- bool(true)
005- int(2)
006- bool(true)