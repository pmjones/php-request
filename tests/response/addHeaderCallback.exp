bool(true)
bool(false)
int(1)
bool(true)
int(2)
bool(true)