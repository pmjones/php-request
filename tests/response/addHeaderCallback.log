
---- EXPECTED OUTPUT
bool(true)
bool(false)
int(1)
bool(true)
int(2)
bool(true)
---- ACTUAL OUTPUT
Fatal error: Uncaught Error: Call to undefined method SapiResponse::getHeaderCallbacks() in /Users/pmjones/Code/pmjones/php-request/tests/response/addHeaderCallback.php:4
Stack trace:
#0 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/tests/response/addHeaderCallback.php on line 4
---- FAILED
