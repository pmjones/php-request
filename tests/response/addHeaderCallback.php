<?php
$func = function($response) {};
$response = new SapiResponse();
var_dump($response->getHeaderCallbacks() === null);
var_dump(array() === $response->getHeaderCallbacks());

$response->addHeaderCallback($func);
var_dump(count($response->getHeaderCallbacks()));
var_dump(array($func) === $response->getHeaderCallbacks());

$response->addHeaderCallback($func);
var_dump(count($response->getHeaderCallbacks()));
var_dump(array($func, $func) === $response->getHeaderCallbacks());
