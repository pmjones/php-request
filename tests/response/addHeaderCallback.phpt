--TEST--
SapiResponse::addHeaderCallback
--CGI--
--INI--
expose_php=0
--FILE--
<?php
$func = function($response) {};
$response = new SapiResponse();
var_dump($response->getHeaderCallbacks() === null);
var_dump(array() === $response->getHeaderCallbacks());

$response->addHeaderCallback($func);
var_dump(count($response->getHeaderCallbacks()));
var_dump(array($func) === $response->getHeaderCallbacks());

$response->addHeaderCallback($func);
var_dump(count($response->getHeaderCallbacks()));
var_dump(array($func, $func) === $response->getHeaderCallbacks());
--EXPECT--
bool(true)
bool(false)
int(1)
bool(true)
int(2)
bool(true)
