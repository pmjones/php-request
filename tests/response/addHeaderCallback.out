Fatal error: Uncaught Error: Call to undefined method SapiResponse::getHeaderCallbacks() in /Users/pmjones/Code/pmjones/php-request/tests/response/addHeaderCallback.php:4
Stack trace:
#0 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/tests/response/addHeaderCallback.php on line 4